clang/test/CXX/basic/basic.start/basic.start.main/p3.cpp

```diff
@@ -78,12 +80,12 @@ namespace ns {
 extern "C" struct A { int main(); }; // ok
 
 namespace c {
-  extern "C" void main(); // expected-warning {{'main' should not be 'extern "C"'}}
+  extern "C" int main(); // expected-warning {{'main' should not be 'extern "C"'}}
 }
 
 extern "C" {
   namespace Z {
-    void main(); // expected-warning {{'main' should not be 'extern "C"'}}
```

Comment:
Why did you change this test?

Reply:
To avoid unnecessary diagnostics and only validate invalid linkage for the main function, since the main function will not be interrupted by invalid usage. Should I revert it?

Reply:
Yes - i think it's useful to test that `void main()` is recognized as a main function, albeit an invalid one

Reply:
@cor3ntin Thanks for the feedback. I've reverted to test _void main_ functions.